src/character.cpp

```diff
@@ -490,8 +490,31 @@ character_id Character::getID() const
 
 void Character::randomize_blood()
 {
-    my_blood_type = static_cast<blood_type>( rng( 0, static_cast<int>( blood_type::num_bt ) - 1 ) );
-    blood_rh_factor = one_in( 2 );
+    // Blood type distribution data is from
```

Comment:
```suggestion
     // Blood type distribution data is from a combination of
     // https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7354354/ and
     // data/json/npcs/appearance_trait_groups.json
    static const std::array<std::tuple<double, blood_type, bool>, 8> blood_type_distribution = {{
            std::make_tuple( 0.3917, blood_type::blood_O, true ),
            std::make_tuple( 0.0397, blood_type::blood_O, false ),
            std::make_tuple( 0.3097, blood_type::blood_A, true ),
            std::make_tuple( 0.0379, blood_type::blood_A, false ),
            std::make_tuple( 0.1493, blood_type::blood_B, true ),
            std::make_tuple( 0.0147, blood_type::blood_B, false ),
            std::make_tuple( 0.0495, blood_type::blood_AB, true ),
            std::make_tuple( 0.0075, blood_type::blood_AB, false )
```

Reply:
Could you briefly demonstrate how you worked out these numbers?

Reply:
> Could you briefly demonstrate how you worked out these numbers?

I'm glad you asked, since I just figured out I was reading the wrong line on the table in the article... the more-correct numbers are:

- A+: 0.338
- A-: 0.0414
- B+: 0.1361
- B-: 0.0134
- AB+: 0.0437
- AB-: 0.0066
- O+: 0.3821
- O-: 0.0387

I worked it out in the below Excel (sigh...) file.
[blood_types.xlsx](https://github.com/CleverRaven/Cataclysm-DDA/files/5823591/blood_types.xlsx)

I first converted the appearance demographics to (very rough!) racial/ethnic ones to match those the study used (PMC reference above). I figured out from the study what the proportions of different blood types were for each of the racial/ethnic groups. I then multiplied these times the group proportions, and added up each group's contribution to each blood type.